### DISCUSSION TITLE
Is there a script to retrieve new changes from Wikipedia?

### DISCUSSION THREAD

**Discussion by 0a1b** [fja05680/sp500#13] [General]
Hi,

first of all, thank you very much for sharing and updating the changes!

I wonder whether you have a script that does the job and if so, whether you could share that as well.

Best

**Comment by fja05680** [comment]
Hello,

Thanks for your kind words.  I run sp500.ipynb which retrieves current S&P 500 composition from Wikipedia and outputs 'sp500.csv'.  Then I run sp500_historical.ipynb which updates the original S&P 500 Historical Components & Changes.csv and outputs S&P 500 Historical Components & Changes(MM-DD-YYYY).csv, but is also shows me the differences between the last line in S&P 500 Historical Components & Changes(MM-DD-YYYY).csv and 'sp500.csv'.  From there I use the changes section in Wikipedia S&P500 and some google searching.  The changes listed in wikipedia are not complete, so it can't be automated.  I hope that answers your question.

Farrell
